_drafts/CocoaPods-0.36.markdown

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+title:  "CocoaPods 0.36 - Framework and Swift Support"
+author: marius
+categories: cocoapods releases
+---
+
+TL;DR: _CocoaPods 0.36_ has been released, with the long-awaited support for Frameworks and Swift.
+
+This initially supports dynamic framework and by that it also brings enhanced support for dependencies using Apple's new programming language Swift. This has been one of the largest singular projects for CocoaPods, affecting almost all of CocoaPods' subsystems like Xcodeproj.
+
+<!-- more -->
+
+
+## And there were Swift & Dynamic Frameworks on iOS
+
+Dynamic frameworks have always been available on OS X. That's different for iOS.
+Apple's mobile platform introduced third-party dynamic framework support in iOS 8.
+So the least common denominator was found before with using static libraries, which are supported on both platforms.
+
+To the same time when Dynamic Frameworks were introduced, Apple also introduced Swift.
+If you have third-party dependencies in Swift, you have only two choices:
+Either throw them in your project and compile one fat binary, which is no practical solution as this increases build times by the lack of incremental compilation and makes it hard to generically manage very different dependencies, which could require different build settings etc. Or you can facilitate frameworks.
+Static libraries are not an option anymore.
+
+Why is that the case? Because Swift is still very open in design and subject to heavy changes. Unlike Objective-C, Apple doesn't ship the Swift standard runtime libraries with iOS.
+This decouples the language version from the platform version.
+When you build an app with Swift, you're responsible yourself to ship them.
+The toolchain provides at this place `swift-stdlib-tool`, which [covers the common use cases](http://samdmarshall.com/blog/swift_and_objc.html).
+As those libraries can't be statically linked multiple times, and not at all in different versions.
+Furthermore it is desirable to embed them only once and not embedded multiple times, because of constraints to memory size and network speed, which are relevant for distribution.
+
+With this release, we initially allow to use both in combination with CocoaPods.
```

Comment:
With this release, we've kept the option to integrate Pods as static libraries, while introducing the ability to integrate as a proper Framework.